docs/concepts/advanced/template_tags.md

```diff
@@ -0,0 +1,197 @@
+---
```

Comment:
Added a documentation page on creating custom template tags that make use of django-component's features.

Here's a preview:


https://github.com/user-attachments/assets/1e1a1dd6-8f89-452a-89a9-bcd76552a897